### DISCUSSION TITLE
Converting battery files between vba-m and no$gba

### DISCUSSION THREAD

**Discussion by hmsong01** [visualboyadvance-m/visualboyadvance-m#1151] [General]
### Expected behavior
Use the Custom Robo GX save file from no$gba (v2.6, 2.9b, 3.00) to vba-m nightbuild

### Actual behavior
It doesn't work.  It just starts a new game.

### Steps to reproduce the behavior
Changed the save format of no$gba to Raw (even before playing the game on no$gba, to produce save file in raw format).
Did factory reset for vba-m.
Renamed the no$gba save file to match the rom for vba-m (same name though).

### ROM details

Which games are affected by the issue (please state if GBA or GB game):
GBA -- Custom Robo GX (J)

Which games, if any, are NOT affected by the issue (please state if GBA or GB game):

Include screenshots of ROM Information (File > ROM Information):
https://i.ibb.co/6tcNpLG/1.png

If you are using a BIOS file, please say which one.

### Option details

Video driver (simple, opengl):

Sound driver (directsound, xaudio, openal, SDL):

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.):
Window 10

If using Linux, specify if using xorg or Wayland:

Version of code (pre-built binary version, or commit ref, or just "master"):

I actually tried to use the Link feature, but it didn't work (Link works in most games, except Custom Robo GX, for whatever reason).

**Comment by Squall-Leonhart** [comment]
supply the save file

**Comment by hmsong01** [comment]
Here:

[Custom Robo GX (J)1.zip](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6066182/Custom.Robo.GX.J.1.zip)

That's for no$gba.  The reason I want to use no$gba file is because there are several unlockable items from having played multiplayers.  And for whatever reason, vba can't multiplay (1 computer) for this particular game.  vba works for all other games.

**Comment by hmsong01** [comment]
I can convert vba --> no$gba save file using this:
https://www.romhacking.net/utilities/416/

But for whatever reason, it doesn't work the other way around, despite the description of the program saying that it can.  At least, not for Custom Robo GX.

**Comment by Squall-Leonhart** [comment]
Game uses Eeprom save type, 
Your nogba exported save is Flash 128KB.

**Comment by hmsong01** [comment]
So... what do I need to do to make the save file usable?  I don't mind restarting the game, but I do need to be able to transfer no$gba save files to vba, so that I actually have multiplayer usage counted towards the save file.  Or is there a way to change something in vba to use the save file?

Obviously, the best option is for me to be able to use vba to link the games, but I assume that's not an option.

**Comment by Squall-Leonhart** [comment]
The save file itself needs converting, 

@rkitover get a load of this
![image](https://user-images.githubusercontent.com/1098176/109609778-f10a9e00-7b7f-11eb-94e7-28d73f0587f6.png)

The bytes are reversed in the nogba save. (left) compared to vbam (right)

either that or we're the wrong order?

**Comment by Squall-Leonhart** [comment]
@hmsong01 
[Custom.Robo.GX.J.1.zip](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6066670/Custom.Robo.GX.J.1.zip)

Here is your VBAM save

To go b ack to VBAM from Nocash, you need to trim everything beyond 8192 with a hex editor (basically the long ass block of y's at the end of the file in hex workshop)

You then need to use the byte flip function to flip everything as 64bit quad.

**Comment by hmsong01** [comment]
First of all, thank you for that save file.  I'm gonna try to figure out what address contains the necessary information to unlock certain items for this game.

Second, that picture sure looks confusing.  Some bytes are reversed, while others aren't.  What the heck?

I used HxH before, but I don't' know what you mean by "8192".  Do you mean the address 00008192?  Because the address doesn't go that far.

Also, could you please elaborate "You then need to use the byte flip function to flip everything as 64bit quad."?  I'm sorry for not understanding what would seem to be a basic for you, but I'd appreciate it if you can be patient with me.  Sorry, and thanks.

**Comment by Squall-Leonhart** [comment]
00002000 is the hex location
delete anything beyond this and your file will become 8192 bytes which is the size of EEPROM_V122

Then under tools > operations > byte flip
![image](https://user-images.githubusercontent.com/1098176/109619925-6cbf1780-7b8d-11eb-8f31-f26d9ac2c082.png)

**Comment by hmsong01** [comment]
Okay, I was able to delete anything from 00002000 and on, and it did make it to 8192 bytes.  But I don't see tool > operations.  

I'm using HxD.  What hex editor are you using?  Or am I using diff version of HxD?
![1](https://user-images.githubusercontent.com/44662260/109640504-ec4fe500-7b93-11eb-9e8e-78fb163d7ddf.png)

Also, is there a way to make Custom Robo GX multiplayer work in vbam?

**Comment by Squall-Leonhart** [comment]
Hex Workshop is a different application

> Also, is there a way to make Custom Robo GX multiplayer work in vbam?

there is a branch that might work, but it requires manual building.

**Comment by hmsong01** [comment]
I'm gonna download Hex Workshop to see if I can work with it.  Btw, is there a way to delete 00002000 and on?  There's a lot of lines, and I'm wonder if there's a faster way to select them all.

> 
> there is a branch that might work, but it requires manual building.

"Manual building"?  What does that mean?  I assume it's more complicated than simply going Options > Gameboy Advance > Configure and changing the save file type to Flash 128K?

In any case, the problem is solved.  Thank you.  I know multiplayer isn't your priority, but I shall hope that vbam will one day be able to play this game on multiplayer.
